### ISSUE TITLE
Attribute error `object has no attribute 'var'`

### ISSUE THREAD

**Issue by WeiN76LQh** [droe/binja-blocks#16]
```
[ScriptingProvider] stack block 181a6e314: AttributeError: 'HighLevelILAssign' object has no attribute 'var'
[ScriptingProvider] Traceback (most recent call last):
[ScriptingProvider]   File "/Users/0xalsr/Library/Application Support/Binary Ninja/plugins/binja-blocks/blocks.py", line 765, in annotate_stack_block_literal
[ScriptingProvider]     for insn in shinobi.yield_struct_field_assign_hlil_instructions_for_var_id(bl.insn.function, bl.insn.var.identifier):
[ScriptingProvider] AttributeError: 'HighLevelILAssign' object has no attribute 'var'
```

This was found running the `Annotate all blocks` command on DSC for iOS 18.1.1 22B91 iPhone17,2 with the `libsystem_c.dylib` (loaded by default), `Foundation` and `libsystem_blocks.dylib` images loaded.

Block instance:
```c
181a6e314                    stack_block_var_1e0.isa = &__NSConcreteStackBlock
181a6e31c                    stack_block_var_1e0.flags = 0xc0000000
181a6e31c                    stack_block_var_1e0.reserved = 0
181a6e324                    int64_t x8_8
181a6e324                    
181a6e324                    if (x21_3 == 0)
181a6e324                        x8_8 = x0_81
181a6e324                    else
181a6e324                        x8_8 = x23_2
181a6e324                    
181a6e344                    stack_block_var_1e0.invoke = ___132-[NSAttributedString(NSAttributedStringFormattingSPI) __initWithFormat:attributeOptions:formattingOptions:locale:context:arguments:]_block_invoke
181a6e344                    stack_block_var_1e0.descriptor = &block_descriptor_1e0fc7070
181a6e348                    stack_block_var_1e0.__offset(0x30).q = x22_1
181a6e348                    stack_block_var_1e0.__offset(0x38).q = x19_2
181a6e34c                    stack_block_var_1e0.__offset(0x40).q = x8_8 - x19_2
181a6e354                    stack_block_var_1e0.byref_ptr_48 = &block_byref_var_198
181a6e35c                    stack_block_var_1e0.strong_ptr_48 = obj_3
```

Block descriptor:
```c
1e0fc7070  struct Block_descriptor_1e0fc7070 block_descriptor_1e0fc7070 = 
1e0fc7070  {
1e0fc7070      uint64_t reserved = 0xc0008000
1e0fc7078      uint64_t size = 0x48
1e0fc7080      char const* signature = 0x182604fd0 {"v40@?0@"NSDictionary"8{_NSRange=QQ}16^B32"}
1e0fc7088      uint64_t layout = 0x110
1e0fc7090  }
```

There's multiple instances of this error and the type of the object that doesn't have the `var` attribute varies. I've seen `HighLevelILAssign`, `HighLevelILLabel` and `HighLevelILCase`. Some checking will need to be implemented to ensure things are as expected.

**Comment by droe** [comment]
Can repro w/macOS 14.x DSC.

**Comment by droe** [comment]
Fixed by 293683e.  More similar fixes to come, but closing this issue.
